src/common/quantile.cc

```diff
@@ -112,6 +144,107 @@ void HostSketchContainer::PushRowPage(SparsePage const &page,
   monitor_.Stop(__func__);
 }
 
+void HostSketchContainer::AllReduce(
```

Comment:
Note: this function is high risk for having a bug due to length, complex algorithm, distributed code and multithreaded code. Consider if this can be separated into steps where the output of each step is verifiable.

Reply:
Spited up.